src/taipy/core/pipeline/_pipeline_manager.py

```diff
@@ -164,13 +236,109 @@ def _submit(
+                scenario_manager._set(scenario)
+
+            if hasattr(cls, "_EVENT_ENTITY_TYPE"):
+                for pipeline_id in pipeline_ids:
+                    _publish_event(cls._EVENT_ENTITY_TYPE, pipeline_id, EventOperation.DELETION, None)
+        except (ModelNotFound, KeyError):
+            cls.__log_error_entity_not_found(pipeline_id)
+            raise ModelNotFound(cls._dir_name, pipeline_id)
+
+    @classmethod
+    def _delete_by_version(cls, version_number: str):
+        """
+        Deletes Pipelines by version number.
+        """
+        for scenario in _ScenarioManagerFactory()._build_manager()._repository._search("version", version_number):
+            cls._delete_many(scenario.pipelines.values())
+
+    @classmethod
+    def _delete(cls, pipeline_id: PipelineId):
+        """
+        Deletes a Pipeline by id.
+        """
+        pipeline_name, scenario_id = cls._breakdown_pipeline_id(pipeline_id)
+
+        if scenario := _ScenarioManagerFactory._build_manager()._get(scenario_id):
+            if pipeline_name in scenario._pipelines.keys():
+                scenario.remove_pipelines([pipeline_name])
+                if hasattr(cls, "_EVENT_ENTITY_TYPE"):
+                    _publish_event(cls._EVENT_ENTITY_TYPE, pipeline_id, EventOperation.DELETION, None)
+                return
+        raise ModelNotFound(cls._dir_name, pipeline_id)
+
+    @classmethod
+    def _exists(cls, entity_id: str) -> bool:
+        """
+        Returns True if the entity id exists.
+        """
+        return True if cls._get(entity_id) else False
+
     @classmethod
     def _hard_delete(cls, pipeline_id: PipelineId):
         pipeline = cls._get(pipeline_id)
         entity_ids_to_delete = cls._get_children_entity_ids(pipeline)
         entity_ids_to_delete.pipeline_ids.add(pipeline.id)
         cls._delete_entities_of_multiple_types(entity_ids_to_delete)
 
+    @classmethod
+    def _export(cls, id: str, folder_path: Union[str, pathlib.Path]):
+        """
+        Export a Pipeline entity.
+        """
+        if isinstance(folder_path, str):
+            folder: pathlib.Path = pathlib.Path(folder_path)
+        else:
+            folder = folder_path
+
+        export_dir = folder / cls._dir_name
+        if not export_dir.exists():
+            export_dir.mkdir(parents=True)
+
+        export_path = export_dir / f"{id}.json"
+        pipeline_name, scenario_id = cls._breakdown_pipeline_id(id)
+        pipeline = {"id": id, "owner_id": scenario_id, "parent_ids": [scenario_id], "name": pipeline_name}
+
+        scenario = _ScenarioManagerFactory._build_manager()._get(scenario_id)
+        if pipeline_data := scenario._pipelines.get(pipeline_name, None):
+            pipeline.update(pipeline_data)
+            with open(export_path, "w", encoding="utf-8") as export_file:
+                export_file.write(json.dumps(pipeline))
+        else:
+            raise ModelNotFound(cls._dir_name, id)
```

Comment:
I think the way we raise `ModelNotFound` in SQLRepository is different, without the dir name.